### PR TITLE
FieldDefinition: Add missing forbidden names

### PR DESCRIPTION
**Pull request by blankse**
`Fatal error: Declaration of Pimcore\Model\DataObject\Test::setClassId(?string $classId) must be compatible with Pimcore\Model\DataObject\Concrete::setClassId($o_classId) in /var/www/html/var/classes/DataObject/Test.php on line 188`

See also https://github.com/pimcore/admin-ui-classic-bundle/pull/545

### CODE REVIEW

File: models/DataObject/ClassDefinition/Data.php [hunk 1]
```diff
@@ -76,33 +76,30 @@ abstract class Data implements DataObject\ClassDefinition\Data\TypeDeclarationSu
...
-        'childrenSortby', 'childrensortorder', 'versioncount', 'dirtylanguages', 'dirtyfields', 'classTitle',
+        'apipluginbroker', 'byid', 'bypath', 'cachetag', 'cachetags', 'childamount', 'children', 'childrensortby',
+        'childrensortorder', 'childs', 'class', 'classid', 'classname', 'classtitle', 'closestparentofclass',
+        'creationdate', 'dao', 'data', 'definition', 'dependencies', 'dirtyfields', 'dirtylanguages', 'fieldname',
+        'fullpath', 'haschildren', 'hassiblings', 'id', 'idpath', 'index', 'key', 'language', 'latestversion',
+        'lazyloadedfieldnames', 'list', 'locked', 'modificationdate', 'nextparentforinheritance',
+        'object', 'omitmandatorycheck', 'parent', 'parentclass', 'parentid', 'path', 'permissions',
+        'permissionsforuser', 'properties', 'property', 'published', 'relationdata', 'resource', 'scheduledtasks',
+        'siblings', 'type', 'usermodification', 'userowner', 'userpermissions', 'value', 'valueforfieldname',
+        'valuefromparent', 'values', 'versioncount', 'versions',
```
Comment (kingjia90) [inline models/DataObject/ClassDefinition/Data.php:89]: Thank you again for your PRs

I've found some more cases that we may have missed, basically some of the getter methods in  https://github.com/pimcore/pimcore/tree/11.x/models/DataObject/AbstractObject.php

realpath, realFullPath, currentfullpath, types(plural which returns CONSTs, the singular one is already covered) ,hideunpublished,getinheritedValues(method starts as getGet), childpermissions

Comment (blankse) [reply models/DataObject/ClassDefinition/Data.php:89]: @kingjia90 I added it: https://github.com/pimcore/pimcore/pull/17109/commits/09ce06b41ad2f118ee23251190f6d41f005fb969

**Comment by robertSt7** [comment]
Hi @blankse I found two missing forbidden names. Could you also please add them? Thanks 
`cachekey`
`listingcachekey`

**Comment by blankse** [comment]
@robertSt7 I added it https://github.com/pimcore/pimcore/pull/17109/commits/858eb5a5346d8a663928d1f5f167adfdc19b4197

**Comment by robertSt7** [comment]
@blankse As always, thanks a lot for your contribution

**Pull request merged**